code/modules/atmospherics/machinery/portable/canister.dm

```diff
@@ -368,6 +374,29 @@
 
 	return TRUE
 
+//attaching or droping the singletank bomb
+/obj/machinery/portable_atmospherics/canister/attackby(obj/item/W, mob/user, params)
```

Comment:
Canisters already hold tanks, this entire part is unnecessary.

Reply:
its for holding the tank and not transferring gas between them and showing it to people

Reply:
It already holds the tank whether or not you open the canister to transfer gas.

Reply:
will this do?